src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/DeleteUserProfileServiceImpl.java

```diff
@@ -30,16 +36,61 @@
     @Autowired
     private AuditService auditService;
 
+    @Autowired
+    private IdamFeignClient idamClient;
+
     @Value("${loggingComponentName}")
     private String loggingComponentName;
 
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse deleteByUserId(String userId) {
+        Response idamResponse = idamClient.deleteUser(userId);
+
+        if (idamResponse.status() == NO_CONTENT.value() || idamResponse.status() == NOT_FOUND.value()) {
+            UserProfile userProfile = validateUserStatus(userId);
```

Comment:
should n't validateUserStatus() method be called before deleting user from idam?

how does it handle scenario where user exists in SIDAM but not in UP?

Reply:
may not necessarily be a review comment but i need to understand how the flow will work in this case?